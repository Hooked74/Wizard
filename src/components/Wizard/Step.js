import React, { memo } from "react";
import styled from "styled-components";

const Button = styled.button`
  flex-shrink: 0;
  border: 10px solid ${props => props.color};
  border-radius: 50px;
  width: 40px;
  height: 40px;
  background: transparent;
  position: relative;
  z-index: 1;
  transition: border-color 0.4s cubic-bezier(0.61, 0, 0.74, 0.05);
  position: relative;
  outline: none;
  cursor: pointer;

  ::before {
    content: attr(data-label);
    position: absolute;
    top: -60px;
    left: 50%;
    white-space: nowrap;
    font-size: 26px;
    transform: translate(-50%);
    color: ${props => props.color};
    font-weight: 500;
    transition: color 0.4s cubic-bezier(0.61, 0, 0.74, 0.05);
  }
`;

export default memo(({ label, index, ...rest }) => (
  <Button data-label={label} data-index={index} {...rest} />
));
