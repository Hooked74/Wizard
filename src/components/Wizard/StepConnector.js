import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const ConnectorWrapper = styled.div`
  background-color: ${props => props.color};
  height: 10px;
  width: 100%;
  position: relative;
`;

const LeftConnector = styled.div`
  background-color: ${props => props.color};
  width: 1px;
  height: 10px;
  position: absolute;
  top: 0px;
  left: -1px;
`;

const RightConnector = styled.div`
  background-color: ${props => props.color};
  width: 1px;
  height: 10px;
  position: absolute;
  top: 0px;
  right: -1px;
`;

const ConnectorFiller = styled.div`
  width: ${props => (props.active ? "100%" : "0")};
  height: 10px;
  background-color: ${props => props.color};
  transition: width 0.4s cubic-bezier(0.25, 0.46, 0.45, 0.94);
`;

export default class StepConnector extends PureComponent {
  static propTypes = {
    active: PropTypes.bool.isRequired,
    activeColor: PropTypes.string.isRequired,
    inactiveColor: PropTypes.string.isRequired,
    onTransitionEnd: PropTypes.func
  };

  static getDerivedStateFromProps(nextProps, prevState) {
    const { active: nextPropsActive, activeColor, inactiveColor } = nextProps;

    if (nextPropsActive !== prevState.active) {
      return {
        active: nextPropsActive,
        leftConnectorColor: activeColor,
        rightConnectorColor: inactiveColor
      };
    }

    return prevState;
  }

  constructor(props) {
    super(props);

    const color = props.active ? props.activeColor : props.inactiveColor;

    this.state = {
      active: props.active,
      leftConnectorColor: color,
      rightConnectorColor: color
    };
  }

  onTransitionEnd = () => {
    const { activeColor, inactiveColor, active, onTransitionEnd } = this.props;
    const state = active
      ? { rightConnectorColor: activeColor }
      : { leftConnectorColor: inactiveColor };

    this.setState(state, () => {
      if (onTransitionEnd) onTransitionEnd();
    });
  };

  render() {
    const { activeColor, inactiveColor, active } = this.props;
    const { leftConnectorColor, rightConnectorColor } = this.state;

    return (
      <ConnectorWrapper color={inactiveColor}>
        <LeftConnector color={leftConnectorColor} />
        <ConnectorFiller
          color={activeColor}
          active={active}
          onTransitionEnd={this.onTransitionEnd}
        />
        <RightConnector color={rightConnectorColor} />
      </ConnectorWrapper>
    );
  }
}
