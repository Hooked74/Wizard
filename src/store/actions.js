import { bindActionCreators } from "redux";
import Types from "./types";

export const actions = {
  updateStep(data, index) {
    return {
      type: Types.UPDATE_STEP,
      payload: { data, index }
    };
  }
};

export default dispatcher => bindActionCreators(actions, dispatcher);
