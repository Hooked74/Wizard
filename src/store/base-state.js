export default {
  steps: [
    {
      active: true,
      label: "Design",
      content:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint."
    },
    {
      active: true,
      label: "Build",
      content:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint."
    },
    {
      active: false,
      label: "Launch",
      content:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint."
    }
  ]
};
