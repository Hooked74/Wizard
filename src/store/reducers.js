import Types from "./types";

export default {
  [Types.UPDATE_STEP](state, action) {
    const { index, data } = action.payload;
    const step = {
      ...state.steps[index],
      ...data
    };
    const steps = state.steps.slice(0);

    steps[index] = step;

    return {
      ...state,
      steps
    };
  }
};
