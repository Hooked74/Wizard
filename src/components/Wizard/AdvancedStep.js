import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import Step from "./Step";
import StepConnector from "./StepConnector";

export default class AdvancedStep extends PureComponent {
  static propTypes = {
    index: PropTypes.number.isRequired,
    active: PropTypes.bool.isRequired,
    label: PropTypes.string.isRequired,
    activeColor: PropTypes.string.isRequired,
    inactiveColor: PropTypes.string.isRequired,
    onTransitionEnd: PropTypes.func.isRequired,
    onClick: PropTypes.func.isRequired
  };

  static getDerivedStateFromProps(nextProps, prevState) {
    const { active: nextPropsActive, activeColor, inactiveColor } = nextProps;

    if (nextPropsActive !== prevState.active) {
      const chunkState = nextPropsActive
        ? {
            onTransitionStepEnd: nextProps.onTransitionEnd,
            onTransitionConnectorEnd: () =>
              prevState.component.setState({ stepColor: activeColor })
          }
        : {
            onTransitionStepEnd: () =>
              prevState.component.setState({ activeConnector: false }),
            onTransitionConnectorEnd: nextProps.onTransitionEnd
          };

      return {
        active: nextPropsActive,
        activeConnector: true,
        stepColor: inactiveColor,
        ...chunkState
      };
    }

    return prevState;
  }

  state = {
    component: this, // readonly
    active: this.props.active,
    activeConnector: this.props.active,
    stepColor: this.props.active
      ? this.props.activeColor
      : this.props.inactiveColor,
    onTransitionStepEnd: void 0,
    onTransitionConnectorEnd: void 0
  };

  render() {
    const { index, label, activeColor, inactiveColor, onClick } = this.props;
    const {
      onTransitionStepEnd,
      stepColor,
      onTransitionConnectorEnd,
      activeConnector
    } = this.state;

    return (
      <>
        <StepConnector
          activeColor={activeColor}
          inactiveColor={inactiveColor}
          active={activeConnector}
          onTransitionEnd={onTransitionConnectorEnd}
        />
        <Step
          index={index}
          label={label}
          color={stepColor}
          onClick={onClick}
          onTransitionEnd={onTransitionStepEnd}
        />
      </>
    );
  }
}
