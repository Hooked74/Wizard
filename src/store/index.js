import { applyMiddleware, createStore } from "redux";
import { createLogger } from "redux-logger";
import thunk from "redux-thunk";
import reducers from "./reducers";

export default baseState => {
  const initialState = {
    steps: [],
    ...baseState
  };

  const reducer = (state, action) =>
    reducers[action.type] ? reducers[action.type](state, action) : state;
  const middlewares = [thunk];

  if (process.env.NODE_ENV === "development") {
    middlewares.push(
      createLogger({
        collapsed: true
      })
    );
  }

  return createStore(reducer, initialState, applyMiddleware(...middlewares));
};
