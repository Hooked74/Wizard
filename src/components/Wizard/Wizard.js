import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import Step from "./Step";
import AdvancedStep from "./AdvancedStep";

const Steps = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding-top: 50px;
`;

const Content = styled.div`
  padding-top: 15px;
`;

export default class Wizard extends PureComponent {
  static MAX_STEPS = 5;
  static MIN_STEPS = 2;

  static defaultProps = {
    activeColor: "#5111df",
    inactiveColor: "#dbdbdb"
  };

  static propTypes = {
    steps: PropTypes.arrayOf(
      PropTypes.shape({
        label: PropTypes.string.isRequired,
        content: PropTypes.string.isRequired,
        active: PropTypes.bool.isRequired
      })
    ).isRequired,
    activeColor: PropTypes.string,
    inactiveColor: PropTypes.string,
    updateStep: PropTypes.func.isRequired
  };

  lockedClickOnStep = false;

  constructor(props) {
    super(props);

    const { length } = props.steps;

    if (length > Wizard.MAX_STEPS) {
      console.warn(
        `Step limit exceeded. Only ${Wizard.MAX_STEPS} steps will be rendered.`
      );
    } else if (length < Wizard.MIN_STEPS) {
      throw new Error(
        "Ð¡annot render component. Make sure the props is correctly."
      );
    }
  }

  get lastActiveStep() {
    const { steps } = this.props;
    let step;

    for (let i = steps.length; i--; ) {
      if (steps[i].active) {
        step = { ...steps[i], index: i };
        break;
      }
    }

    if (!step) step = { ...steps[0], index: 0 };

    return step;
  }

  onClickStep = e => {
    const clickedStepIndex = parseInt(e.target.getAttribute("data-index"), 10);
    const lastActiveStep = this.lastActiveStep;

    if (
      !this.lockedClickOnStep &&
      Math.abs(clickedStepIndex - lastActiveStep.index) === 1
    ) {
      const { updateStep } = this.props;
      if (clickedStepIndex < lastActiveStep.index) {
        updateStep({ active: false }, lastActiveStep.index);
      } else {
        updateStep({ active: true }, clickedStepIndex);
      }
      this.lockedClickOnStep = true;
    }
  };

  onTransitionStepEnd = () => {
    this.lockedClickOnStep = false;
  };

  render() {
    const { steps, activeColor, inactiveColor } = this.props;

    return (
      <div>
        <Steps>
          <Step
            label={steps[0].label}
            color={activeColor}
            index={0}
            onClick={this.onClickStep}
            onTransitionEnd={this.onTransitionStepEnd}
          />
          {steps.slice(1, Wizard.MAX_STEPS).map((step, index) => (
            <AdvancedStep
              key={index}
              index={index + 1}
              label={step.label}
              active={step.active}
              activeColor={activeColor}
              inactiveColor={inactiveColor}
              onClick={this.onClickStep}
              onTransitionEnd={this.onTransitionStepEnd}
            />
          ))}
        </Steps>
        <Content>{this.lastActiveStep.content}</Content>
      </div>
    );
  }
}
