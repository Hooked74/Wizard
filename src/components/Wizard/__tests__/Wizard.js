import Enzyme, { mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";
import Wizard from "../Wizard";
import baseState from "../../../store/base-state";
import createStore from "../../../store";
import { actions } from "../../../store/actions";

Enzyme.configure({ adapter: new Adapter() });

describe("<Wizard />", () => {
  test("should compare snapshot", () => {
    const wrapper = shallow(
      <Wizard steps={baseState.steps} updateStep={() => void 0} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("simulates click events", () => {
    const store = createStore(baseState);
    let lastActiveIndex = store
      .getState()
      .steps.map(step => step.active)
      .lastIndexOf(true);

    if (!~lastActiveIndex) lastActiveIndex = 0;

    const wrapper = shallow(
      <Wizard
        steps={baseState.steps}
        updateStep={(...args) => store.dispatch(actions.updateStep(...args))}
      />
    );

    // There is no possibility to simulate a click as the mount does not work with React.memo
    wrapper.instance().onClickStep({
      target: {
        getAttribute() {
          return !lastActiveIndex ? 1 : lastActiveIndex - 1;
        }
      }
    });

    let currentLastActiveIndex = store
      .getState()
      .steps.map(step => step.active)
      .lastIndexOf(true);

    if (!~currentLastActiveIndex) currentLastActiveIndex = 0;

    if (lastActiveIndex) {
      expect(currentLastActiveIndex).toBe(lastActiveIndex - 1);
    } else {
      expect(currentLastActiveIndex).toBe(1);
    }
  });
});
