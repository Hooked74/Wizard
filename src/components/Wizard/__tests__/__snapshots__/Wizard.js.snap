// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wizard /> should compare snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wizard
    activeColor="#5111df"
    inactiveColor="#dbdbdb"
    steps={
      Array [
        Object {
          "active": true,
          "content": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint.",
          "label": "Design",
        },
        Object {
          "active": true,
          "content": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint.",
          "label": "Build",
        },
        Object {
          "active": false,
          "content": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint.",
          "label": "Launch",
        },
      ]
    }
    updateStep={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef>
          <UNDEFINED
            color="#5111df"
            index={0}
            label="Design"
            onClick={[Function]}
            onTransitionEnd={[Function]}
          />
          <AdvancedStep
            active={true}
            activeColor="#5111df"
            inactiveColor="#dbdbdb"
            index={1}
            label="Build"
            onClick={[Function]}
            onTransitionEnd={[Function]}
          />
          <AdvancedStep
            active={false}
            activeColor="#5111df"
            inactiveColor="#dbdbdb"
            index={2}
            label="Launch"
            onClick={[Function]}
            onTransitionEnd={[Function]}
          />
        </ForwardRef>,
        <ForwardRef>
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint.
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <UNDEFINED
              color="#5111df"
              index={0}
              label="Design"
              onClick={[Function]}
              onTransitionEnd={[Function]}
            />,
            Array [
              <AdvancedStep
                active={true}
                activeColor="#5111df"
                inactiveColor="#dbdbdb"
                index={1}
                label="Build"
                onClick={[Function]}
                onTransitionEnd={[Function]}
              />,
              <AdvancedStep
                active={false}
                activeColor="#5111df"
                inactiveColor="#dbdbdb"
                index={2}
                label="Launch"
                onClick={[Function]}
                onTransitionEnd={[Function]}
              />,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": "#5111df",
              "index": 0,
              "label": "Design",
              "onClick": [Function],
              "onTransitionEnd": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "type": [Function],
            },
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "active": true,
              "activeColor": "#5111df",
              "inactiveColor": "#dbdbdb",
              "index": 1,
              "label": "Build",
              "onClick": [Function],
              "onTransitionEnd": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "active": false,
              "activeColor": "#5111df",
              "inactiveColor": "#dbdbdb",
              "index": 2,
              "label": "Launch",
              "onClick": [Function],
              "onTransitionEnd": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding-top: 50px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint.",
        },
        "ref": null,
        "rendered": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint.",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bZQynM",
            "isStatic": true,
            "rules": Array [
              "
  padding-top: 15px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bZQynM",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef>
            <UNDEFINED
              color="#5111df"
              index={0}
              label="Design"
              onClick={[Function]}
              onTransitionEnd={[Function]}
            />
            <AdvancedStep
              active={true}
              activeColor="#5111df"
              inactiveColor="#dbdbdb"
              index={1}
              label="Build"
              onClick={[Function]}
              onTransitionEnd={[Function]}
            />
            <AdvancedStep
              active={false}
              activeColor="#5111df"
              inactiveColor="#dbdbdb"
              index={2}
              label="Launch"
              onClick={[Function]}
              onTransitionEnd={[Function]}
            />
          </ForwardRef>,
          <ForwardRef>
            Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint.
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <UNDEFINED
                color="#5111df"
                index={0}
                label="Design"
                onClick={[Function]}
                onTransitionEnd={[Function]}
              />,
              Array [
                <AdvancedStep
                  active={true}
                  activeColor="#5111df"
                  inactiveColor="#dbdbdb"
                  index={1}
                  label="Build"
                  onClick={[Function]}
                  onTransitionEnd={[Function]}
                />,
                <AdvancedStep
                  active={false}
                  activeColor="#5111df"
                  inactiveColor="#dbdbdb"
                  index={2}
                  label="Launch"
                  onClick={[Function]}
                  onTransitionEnd={[Function]}
                />,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "color": "#5111df",
                "index": 0,
                "label": "Design",
                "onClick": [Function],
                "onTransitionEnd": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.memo),
                "compare": null,
                "type": [Function],
              },
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "active": true,
                "activeColor": "#5111df",
                "inactiveColor": "#dbdbdb",
                "index": 1,
                "label": "Build",
                "onClick": [Function],
                "onTransitionEnd": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "active": false,
                "activeColor": "#5111df",
                "inactiveColor": "#dbdbdb",
                "index": 2,
                "label": "Launch",
                "onClick": [Function],
                "onTransitionEnd": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-EHOje",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding-top: 50px;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-EHOje",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint.",
          },
          "ref": null,
          "rendered": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Aliquid soluta eum, possimus veniam reprehenderit ipsam laudantium voluptatibus debitis, deleniti doloribus illum iusto atque consequatur ullam sapiente adipisci officia blanditiis sint.",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bZQynM",
              "isStatic": true,
              "rules": Array [
                "
  padding-top: 15px;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bZQynM",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
